### PR TITLE
XML documentation for Microsoft.Build

### PR DESCRIPTION
**Pull request by jeffkl**
* Also found a mistake in Microsoft.Build.Framework.ProjectStartedEventArgs documentation

### CODE REVIEW

File: src/Shared/AssemblyNameExtension.cs [hunk 1]
```diff
@@ -671,7 +669,6 @@ internal bool Equals(AssemblyNameExtension that, bool considerRetargetableFlag)
         /// <summary>
         /// Compare two assembly names for equality.
         /// </summary>
-        /// <param name="that"></param>
         /// <returns></returns>
```
Comment (rainersigwald) [inline src/Shared/AssemblyNameExtension.cs]: Also remove `returns`?

File: src/Shared/Compat/parametermodifier.cs [hunk 2]
```diff
@@ -13,8 +13,9 @@ namespace System.Reflection
 {  
     using System;
 
+
```
Comment (rainersigwald) [inline src/Shared/Compat/parametermodifier.cs]: New blank line?

File: src/Shared/INodePacketTranslator.cs [hunk 3]
```diff
@@ -8,11 +8,7 @@
 using System;
 using System.Collections.Generic;
 using System.Globalization;
-using System.Text;
 using System.IO;
-using Microsoft.Build.Collections;
-using Microsoft.Build.Execution;
-using Microsoft.Build.Shared;
```
Comment (rainersigwald) [inline src/Shared/INodePacketTranslator.cs]: We've mostly been avoiding `using` cleanups while we're in two branches, but no need to undo this . . .

File: src/XMakeBuildEngine/BackEnd/Components/BuildRequestEngine/BuildRequestEngine.cs [hunk 4]
```diff
@@ -1288,6 +1288,7 @@ private void IssueBuildRequest(BuildRequestBlocker blocker)
         /// Queue an action to be run in the engine.
         /// </summary>
         /// <param name="action">The action to execute.</param>
+        /// <param name="isLastTask"><code>true</code> if this is the last task, otherwise <code>false</code>.</param>
```
Comment (rainersigwald) [inline src/XMakeBuildEngine/BackEnd/Components/BuildRequestEngine/BuildRequestEngine.cs]: the last task _for this queue_.

File: src/XMakeBuildEngine/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupIntrinsicTask.cs [hunk 5]
```diff
@@ -137,6 +137,8 @@ internal override void ExecuteTask(Lookup lookup)
         /// </summary>
         /// <param name="child">The item specification to evaluate and add.</param>
         /// <param name="bucket">The batching bucket.</param>
+        /// <param name="keepMetadata">An <see cref="ISet{String}"/> of metadata to keep.</param>
+        /// <param name="removeMetadata">An <see cref="ISet{String}"/> of metadata to remove.</param>
```
Comment (rainersigwald) [inline src/XMakeBuildEngine/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupIntrinsicTask.cs]: This is metadata _names_, right?

File: src/XMakeBuildEngine/Construction/Solution/ProjectInSolution.cs [hunk 6]
```diff
@@ -22,12 +22,30 @@ namespace Microsoft.Build.Construction
...
+        /// </summary>
+        Unknown,
+        /// <summary>
+        /// C#, VB, and VJ# projects
+        /// </summary>
+        KnownToBeMSBuildFormat,
+        /// <summary>
+        /// 
+        /// </summary>
+        SolutionFolder,       // Not really a project, but persisted as such in the .SLN file.
```
Comment (rainersigwald) [inline src/XMakeBuildEngine/Construction/Solution/ProjectInSolution.cs]: Can you move this comment to the doc comment?

File: src/XMakeBuildEngine/Utilities/EngineFileUtilities.cs [hunk 7]
```diff
@@ -94,6 +95,8 @@ private static bool FilespecHasWildcards(string filespecEscaped)
         /// </summary>
         /// <param name="directoryEscaped">The directory to evaluate, escaped.</param>
         /// <param name="filespecEscaped">The filespec to evaluate, escaped.</param>
+        /// <param name="returnEscaped"><code>true</code> to return escaped specs.</param>
+        /// <param name="excludeSpecsUnescaped"></param>
```
Comment (rainersigwald) [inline src/XMakeBuildEngine/Utilities/EngineFileUtilities.cs]: "The exclude specification, unescaped."

File: src/Shared/Compat/parametermodifier.cs (hunk 2 above)
Comment (jeffkl) [inline src/Shared/Compat/parametermodifier.cs]: I meant to revert this file.  This is now done and I force pushed to get rid of this change.

File: src/Shared/INodePacketTranslator.cs (hunk 3 above)
Comment (jeffkl) [inline src/Shared/INodePacketTranslator.cs]: Ok good to know

File: src/Shared/AssemblyNameExtension.cs (hunk 1 above)
Comment (jeffkl) [inline src/Shared/AssemblyNameExtension.cs]: Done

File: src/XMakeBuildEngine/Construction/Solution/ProjectInSolution.cs (hunk 6 above)
Comment (jeffkl) [inline src/XMakeBuildEngine/Construction/Solution/ProjectInSolution.cs]: The comment seemed really "internal" and not something we'd want showing up in intellisense.  Do you want to reword it to sound more customer-facing?

File: src/XMakeBuildEngine/Utilities/EngineFileUtilities.cs (hunk 7 above)
Comment (jeffkl) [inline src/XMakeBuildEngine/Utilities/EngineFileUtilities.cs]: Done

File: src/XMakeBuildEngine/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupIntrinsicTask.cs (hunk 5 above)
Comment (jeffkl) [inline src/XMakeBuildEngine/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupIntrinsicTask.cs]: Updated all occurrences

File: src/XMakeBuildEngine/BackEnd/Components/BuildRequestEngine/BuildRequestEngine.cs (hunk 4 above)
Comment (jeffkl) [inline src/XMakeBuildEngine/BackEnd/Components/BuildRequestEngine/BuildRequestEngine.cs]: Done

File: src/XMakeBuildEngine/Construction/Solution/ProjectInSolution.cs (hunk 6 above)
Comment (rainersigwald) [inline src/XMakeBuildEngine/Construction/Solution/ProjectInSolution.cs]: Solution folders appear in the `.sln` file, but aren't buildable projects.

?

Comment (jeffkl) [inline src/XMakeBuildEngine/Construction/Solution/ProjectInSolution.cs]: Done

I :heart: these new GitHub features (like being able to reply to a comment)
